session.go

```diff
@@ -31,3 +31,30 @@ type Session interface {
 	// session implementation.
 	Version() (msize int, version string)
 }
+
+// WriteFull writes the whole p slice to the specified fid
+func WriteFull(s Session, ctx context.Context, fid Fid, p []byte, offset int64) error {
+	for len(p) > 0 {
+		written, err := s.Write(ctx, fid, p, offset)
+		if err != nil {
+			return err
+		}
+		p = p[written:]
+		offset += int64(written)
+	}
+	return nil
+}
+
+// ReadFull reads the whole file and returns it in a slice
+func ReadFull(s Session, ctx context.Context, fid Fid, offset int64) (p []byte, err error) {
```

Comment:
After some thought here, it would be much better to implement an `io.Reader` or `io.ReadSeeker` that tracks offset and fid. This way, the `io` utility functions can be used composably. The singature would be something like this:

```
NewReadSeeker(ctx context.Context, s Session, fid Fid) (io.ReadSeeker, error)
```

This should open up correct buffering.